src/art.rs

```diff
@@ -19,33 +19,45 @@ const NODE48MAX: usize = 48;
 
 // Minimum and maximum number of children for Node256
 const NODE256MIN: usize = NODE48MAX + 1;
-const NODE256MAX: usize = 256;
 
 // From the specification: Adaptive Radix tries consist of two types of nodes:
 // Inner nodes, which map prefix(prefix) keys to other nodes,
 // and leaf nodes, which store the values corresponding to the key.
-struct Node<P: Prefix + Clone, V> {
+struct Node<P: Prefix + Clone, V: Clone> {
     pub node_type: NodeType<P, V>, // Type of the node
 }
 
-enum NodeType<P: Prefix + Clone, V> {
+enum NodeType<P: Prefix + Clone, V: Clone> {
```

Comment:
Not sure if every node here should be an Arc<Node<P,V>>, given that the root already has a reference to the root node using Arc.